examples/addsvc/cmd/addsvc/addsvc.go

```diff
@@ -135,11 +136,12 @@ func main() {
 	// the interfaces that the transports expect. Note that we're not binding
 	// them to ports or anything yet; we'll do that next.
 	var (
-		service      = addservice.New(logger, ints, chars)
-		endpoints    = addendpoint.New(service, logger, duration, tracer)
-		httpHandler  = addtransport.NewHTTPHandler(endpoints, tracer, logger)
-		grpcServer   = addtransport.NewGRPCServer(endpoints, tracer, logger)
-		thriftServer = addtransport.NewThriftServer(endpoints)
+		service        = addservice.New(logger, ints, chars)
+		endpoints      = addendpoint.New(service, logger, duration, tracer)
+		httpHandler    = addtransport.NewHTTPHandler(endpoints, tracer, logger)
+		grpcServer     = addtransport.NewGRPCServer(endpoints, tracer, logger)
+		thriftServer   = addtransport.NewThriftServer(endpoints)
+		jsonrpcHandler = addtransport.NewJSONRPCHandler(endpoints, logger)
```

Comment:
Would like to see tracing added to this transport too in the addsvc example just like HTTP and gRPC.

Reply:
I'll need to ponder how best to do this. As it stands, ServerBefore functions are called prior to decoding the request body. This will mean that we can only trace a request to the JSON RPC endpoint; we won't know which method was called, which would hide some useful detail.

I could move the ServerBefore calls to after decoding the JSON, but I'm not sure if/how this would affect other ServerBefore uses, as I'm not really familiar with the use cases for those.

I suppose the other option would be to introduce a tracing-specific option, but that feels a bit icky. Suggestions welcome!

Reply:
Not needed, I know how to plug into existing framework.. I'll add tracing independent from this PR.

Reply:
Great. Thank you.